libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/camera/DynamicCamera.java

```diff
@@ -107,6 +107,9 @@ private double createTilt(double distanceRemaining) {
    */
   private double createZoom(RouteInformation routeInformation) {
     CameraPosition position = createCameraPosition(routeInformation.location(), routeInformation.routeProgress());
+    if (position == null) {
```

Comment:
I'm still unsure how will we end up getting null ref returned by `createCameraPosition`?

Reply:
https://github.com/mapbox/mapbox-navigation-android/blob/d382c3ab6de763f870f2f0202ab77e1ba7bd595f/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/camera/DynamicCamera.java#L151 `MapboxMap#getCameraForLatLngBounds` could return `null` if the map is not visible - this is obviously as designed because why would you want to interact with it if not present?

That's why I remarked the importance of using the lifecycle methods from `NavigationCamera` and `NavigationMapboxMap` i.e. `DynamicCamera` is a lifecycle aware component which doesn't mean anything if the map is not foregrounded. Does that make sense?

Maybe we should remove the `null` check altogether and work on a solution in which these classes are harder to be misused. The challenge here is that `NavigationMapboxMap` and `NavigationCamera` go hand by hand so it's not trivial / simple.

Reply:
imho in this case returning a valid ref of `CameraPosition` that represents an invalid state of camera is better than returning a `null` but i don't know all under the hood details, so i can't say if it's feasible.

Reply:
Yeah, I know what you mean, creating a `CameraPosition` Null Object would be ideal. There's this [`CameraPosition#DEFAULT`](https://github.com/mapbox/mapbox-gl-native/blob/da69a88c04fcd7e93507dde74ffbc4e0fbd57210/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/camera/CameraPosition.java#L21) constant that we could use here. Great idea! Although we cannot get rid off the `null` check completely because [`MapboxMap#getCameraForLatLngBounds`](https://github.com/mapbox/mapbox-gl-native/blob/da69a88c04fcd7e93507dde74ffbc4e0fbd57210/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java#L1420-L1422) is `@Nullable`, we'd need to move it to right before returning the `CameraPosition` https://github.com/mapbox/mapbox-navigation-android/blob/156e6a7aafe3bb603ac7c0ba28a322e0925b7a8d/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/camera/DynamicCamera.java#L148 ¯\\\_(ツ)\_/¯